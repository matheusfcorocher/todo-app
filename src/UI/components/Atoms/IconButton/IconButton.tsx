import React, { useState } from 'react';
import '../../../../App';
import './icon-button.css';

interface IconButtonProps {    
    handleFunction: () => void,
    children: React.ReactNode,
    className?: string,
    title?: string,
}

function IconButton({
    handleFunction,
    children,
    className,
    title
}: IconButtonProps) {
    return (
        <button
            title={title}
            className={`icon-button ${className}`}
            onClick={() => handleFunction()}
            role="button"
        >
            {children}
        </button>
    );
}

export default IconButton;
