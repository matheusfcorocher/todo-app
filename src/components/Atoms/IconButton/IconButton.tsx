import React, { useState } from 'react';
import '../../../App';
import './icon-button.css';

interface IconButtonProps {    
    handleFunction: () => void,
    children: React.ReactNode,
    className?: string,
}

function IconButton({
    handleFunction,
    children,
    className
}: IconButtonProps) {
    return (
        <button
            className={`icon-button ${className}`}
            onClick={() => handleFunction}
            role="button"
        >
            {children}
        </button>
    );
}

export default IconButton;
