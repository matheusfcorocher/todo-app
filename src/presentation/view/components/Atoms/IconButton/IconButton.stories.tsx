import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import IconButton from './IconButton';
import { Default as DeleteIcon } from '../icons/DeleteIcon/DeleteIcon.stories';

export default {
  title: 'Atoms/IconButton',
  component: IconButton,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0 }}>
        <Story />
      </ul>
    ),
  ]
} as ComponentMeta<typeof IconButton>;

const Template: ComponentStory<typeof IconButton> = (args) => 
  <IconButton handleFunction={function (): void {
    throw new Error('Function not implemented.');
  } } children={undefined} />;

export const Default = Template.bind({});
Default.args = {};

export const DeleteButton : ComponentStory<typeof IconButton> = (args) => 
<IconButton handleFunction={function (): void {
throw new Error('Function not implemented.');
} }>
  <DeleteIcon {...DeleteIcon.args}/>
</IconButton>;
