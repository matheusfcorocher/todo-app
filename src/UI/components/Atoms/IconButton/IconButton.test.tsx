import React, { useState } from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import IconButton from './IconButton';

describe("::Components ::Atoms ::IconButton", () => {
    test('renders IconButton', () => {
        render(<IconButton handleFunction={function (): void {
            throw new Error('Function not implemented.');
        } } children={undefined} />);
        const icon = screen.getByRole("button");
        expect(icon).toMatchSnapshot();
    })
})
