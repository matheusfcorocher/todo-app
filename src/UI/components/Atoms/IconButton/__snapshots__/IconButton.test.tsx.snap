// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`::Components ::Atoms ::IconButton renders IconButton 1`] = `
<button
  class="icon-button undefined"
  role="button"
/>
`;
